pkg/scheduler/framework/plugins/availablereplicas/available_replicas.go

```diff
@@ -0,0 +1,80 @@
+package availablereplicas
+
+import (
+	"context"
+	"math"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	estimatorclient "github.com/karmada-io/karmada/pkg/estimator/client"
+	"github.com/karmada-io/karmada/pkg/scheduler/framework"
+)
+
+const (
+	// Name is the name of the plugin used in the plugin registry and configurations.
+	Name = "AvailableReplicas"
+)
+
+// AvailableReplicas is a scoring plugin that favors well-resourced clusters.
+type AvailableReplicas struct {
+	availableReplicasCache []workv1alpha2.TargetCluster
+}
+
+var _ framework.ScorePlugin = &AvailableReplicas{}
+
+// New instantiates the AvailableReplicas plugin.
+func New() (framework.Plugin, error) {
+	return &AvailableReplicas{}, nil
+}
+
+// Name returns the plugin name.
+func (p *AvailableReplicas) Name() string {
+	return Name
+}
+
+// Score calculates the score on the candidate cluster.
+// Get AvailableReplicas according to RB's ReplicaRequirements, and normalize it to the range of 0-100 interval,
+// and convert it into a score.
+func (p *AvailableReplicas) Score(ctx context.Context,
+	spec *workv1alpha2.ResourceBindingSpec, clusters []*clusterv1alpha1.Cluster, name string) (int64, *framework.Result) {
+	strategy := spec.Placement.ReplicaScheduling
+
+	// If the replica division preference is 'Duplicated', ignore the information about cluster available resource.
+	if strategy == nil || strategy.ReplicaSchedulingType == policyv1alpha1.ReplicaSchedulingTypeDuplicated {
+		return framework.MinClusterScore, framework.NewResult(framework.Success)
+	}
+	if p.availableReplicasCache == nil {
+		p.availableReplicasCache = estimatorclient.CalcAvailableReplicas(clusters, spec)
```

Comment:
If `availableReplicasCache` is initialized, I didn't find any place to reset or re-init it, so the if the resource state changes, `availableReplicasCache` will be the same as the last time.

Reply:
> If availableReplicasCache is initialized, I didn't find any place to reset or re-init it

In 78 line.

Reply:
I see, thanks

Reply:
We can consider adding an interface to the `estimator` to calculate the available replicas of a single cluster each time.